packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts

```diff
@@ -0,0 +1,477 @@
+   *
+   * For example, you can change the path to "/new/#{path}", the hostname to
+   * "example.#{host}", or the query to "#{query}&value=xyz".
+   */
+  public static redirect(options: RedirectOptions): ApplicationListenerAction {
+    if ([options.host, options.path, options.port, options.protocol, options.query].findIndex(x => x !== undefined) === -1) {
+      throw new Error('To prevent redirect loops, set at least one of \'protocol\', \'host\', \'port\', \'path\', or \'query\'.');
+    }
+
+    return new ApplicationListenerAction({
+      type: 'redirect',
+      redirectConfig: {
+        statusCode: options.permanent ? 'HTTP_301' : 'HTTP_302',
+        host: options.host,
+        path: options.path,
+        port: options.port,
+        protocol: options.protocol,
+        query: options.query,
+      },
+    });
+  }
+
+  /**
+   * Create an instance of ApplicationListenerAction
+   *
+   * The default class should be good enough for most cases and
+   * should be created by using one of the static factory functions,
+   * but allow overriding to make sure we allow flexibility for the future.
+   */
+  protected constructor(private readonly actionJson: CfnListener.ActionProperty, protected readonly next?: ApplicationListenerAction) {
+  }
+
+  /**
+   * Render the actions in this chain
+   */
+  public renderActions(): CfnListener.ActionProperty[] {
+    return this.renumber([this.actionJson, ...this.next?.renderActions() ?? []]);
+  }
+
+  /**
+   * Called when the action is being used in a listener
+   */
+  public bindToListener(scope: Construct, listener: IApplicationListener, associatingConstruct?: IConstruct) {
+    // Empty on purpose
+    Array.isArray(scope);
+    Array.isArray(listener);
+    Array.isArray(associatingConstruct);
+  }
+
+  /**
+   * Renumber the "order" fields in the actions array.
+   *
+   * We don't number for 0 or 1 elements, but otherwise number them 1...#actions
+   * so ELB knows about the right order.
+   *
+   * Do this in `ApplicationListenerAction` instead of in `Listener` so that we give
+   * users the opportunity to override by subclassing and overriding `renderActions`.
+   */
+  protected renumber(actions: CfnListener.ActionProperty[]): CfnListener.ActionProperty[] {
+    if (actions.length < 2) { return actions; }
+
+    return actions.map((action, i) => ({ ...action, order: i + 1 }));
+  }
+}
+
+/**
+ * Options for `ApplicationListenerAction.forward()`
+ *
+ * @experimental
+ */
+export interface ForwardOptions {
+  /**
+   * The list of target groups to forward to
+   */
+  readonly targetGroups: IApplicationTargetGroup[];
+
+  /**
+   * For how long clients should be directed to the same target group
+   *
+   * Range between 1 second and 7 days.
+   *
+   * @default - No stickiness
+   */
+  readonly stickinessDuration?: Duration;
+}
+
+/**
+ * Options for `ApplicationListenerAction.weightedForward()`
+ *
+ * @experimental
+ */
+export interface WeightedForwardOptions {
```

Comment:
Why is this a different type from `ForwardOptions`?

Reply:
One just takes a list of target groups (common case), the other one takes a list of structures of targetgroups and weights (more complex, only when you need it).